streaming/src/main/scala/org/apache/spark/streaming/receiver/ReceivedBlockHandler.scala

```diff
@@ -151,13 +167,21 @@ private[streaming] class WriteAheadLogBasedBlockHandler(
    */
   def storeBlock(blockId: StreamBlockId, block: ReceivedBlock): ReceivedBlockStoreResult = {
 
+    var numRecords = None: Option[Long]
     // Serialize the block so that it can be inserted into both
     val serializedBlock = block match {
       case ArrayBufferBlock(arrayBuffer) =>
-        blockManager.dataSerialize(blockId, arrayBuffer.iterator)
+        val countIterator = new CountingIterator(arrayBuffer.iterator)
```

Comment:
For ArrayBufferBlock, isnt it just easier to just get the count directly rather than going through the CountingIterator again? Why even add the slightly inefficiency of going through the counting iterator?